kedro-org/kedro-viz
# Add disable preview toggle in settings panel

## Description
Related to: #1921

This feature enables users to control data previews for all nodes from the settings panel of Kedro-Viz when running kedro viz locally.

After running Kedro-Viz locally, dataset previews for all nodes will be enabled by default. However, users will have the ability to enable or disable this feature from the Kedro-Viz settings panel.

## Development Notes

**Backend Changes:**

- Introduced a new API endpoint `/api/preferences` to handle user preferences updates.
- Created a `UserPreference` model to define the preference schema.
- Implemented a new function in `router.py` to update the `showDatasetPreviews` preference and set the class variable accordingly.
- Adjusted the logic in `responses.py` to use the new preference for determining dataset preview settings.

**Frontend Changes:**

- Added a new `showDatasetPreviews` toggle in the Redux state to manage the setting.
- Updated the `SettingsModal` component to include the `showDatasetPreviews` toggle.

**How to Test:**

- Run kedro viz locally via
- Open the Kedro-Viz settings panel.
- Toggle the Dataset Previews setting on and off.
- Verify that the dataset previews for all nodes are enabled/disabled based on the setting.
- Check the network tab to confirm that Preferences API is called and that the preferences are correctly updated on the backend. (Might need to toggle preserve logs to view it as saving in the settings menu refreshes the page and clears network logs)

New dataset preview toggle:

![image](https://github.com/user-attachments/assets/2d8b38da-be3c-4818-8e28-b454789e2e6c)

New API endpoint:

<img width="848" alt="image" src="https://github.com/user-attachments/assets/01a26ec9-9fec-4666-8759-d4ebd2a5617d">


## QA notes



## Checklist
